### PR TITLE
Repetition for unquotation

### PR DESCRIPTION
**Pull request by pczarn**

### CODE REVIEW

File: quasi_tests/tests/test.rs.in [hunk 1]
```diff
@@ -277,3 +277,129 @@ fn test_stmt_semicolons() {
...
+    exprs.push(quote_expr!(&cx, 1 + 2));
+    exprs.push(quote_expr!(&cx, 2 * 3));
+    let exprs = exprs.into_iter();
+
+    let expr = quote_expr!(&cx,
+        ( $($exprs),* )
+    );
+
+    assert_eq!(pprust::expr_to_string(&expr), "(1 + 2, 2 * 3)");
+}
```
Comment (pczarn) [inline quasi_tests/tests/test.rs.in:298]: Examples are here, in tests.

**Comment by pczarn** [comment]
For now, I'd like to add repetition to quasi, not to rustc's libsyntax, because prototyping is easier with aster and in a small repository.

**Comment by pczarn** [comment]
@erickt could you review and merge?

File: quasi_tests/tests/test.rs.in [hunk 2]
```diff
@@ -277,3 +277,129 @@ fn test_stmt_semicolons() {
...
+#[test]
+#[should_panic]
+fn test_fail_unbounded_repetition() {
+    let sess = parse::ParseSess::new();
+    let mut feature_gated_cfgs = Vec::new();
+    let cx = make_ext_ctxt(&sess, &mut feature_gated_cfgs);
+
+    let unit = ();
+
+    quote_expr!(&cx,
```
Comment (erickt) [inline quasi_tests/tests/test.rs.in:372]: Could you test the value here? I'm guessing it's `assert_eq!(pprust::expr_to_string(&expr), "()");` but it'd be nice to be sure.

edit: oops, I didn't see it was a `#[should_panic]`, nevermind.

**Pull request merged**